test/integration/xds_delegate_extension_integration_test.cc

```diff
@@ -0,0 +1,236 @@
+                port_value: 0
+layered_runtime:
+  layers:
+  - name: some_static_layer
+    static_layer:
+      foo: whatevs
+      bar: yar
+  - name: some_rtds_layer
+    rtds_layer:
+      name: some_rtds_layer
+      rtds_config:
+        resource_api_version: V3
+        api_config_source:
+          api_type: GRPC
+          transport_api_version: V3
+          grpc_services:
+            envoy_grpc:
+              cluster_name: rtds_cluster
+          set_node_on_first_message_only: true
+  - name: some_admin_layer
+    admin_layer: {{}}
+xds_delegate_extension:
+  name: envoy.config.xds.test_delegate
+  typed_config:
+    "@type": type.googleapis.com/test.envoy.config.xds.TestXdsResourcesDelegateConfig
+admin:
+  access_log:
+  - name: envoy.access_loggers.file
+    typed_config:
+      "@type": type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog
+      path: "{}"
+  address:
+    socket_address:
+      address: 127.0.0.1
+      port_value: 0
+)EOF",
+                     Platform::null_device_path);
+}
+
+class XdsDelegateExtensionIntegrationTest : public Grpc::GrpcClientIntegrationParamTest,
+                                            public HttpIntegrationTest {
+public:
+  XdsDelegateExtensionIntegrationTest()
+      : HttpIntegrationTest(Http::CodecType::HTTP2, ipVersion(), bootstrapConfig()) {
+    // TODO(abeyad): Add test for Unified SotW when the UnifiedMux support is implemented.
+    use_lds_ = false;
+    create_xds_upstream_ = true;
+  }
+
+  void TearDown() override {
+    if (xds_connection_ != nullptr) {
+      cleanUpXdsConnection();
+    }
+  }
+
+  void initialize() override {
+    // The tests infra expects the xDS server to be the second fake upstream, so
+    // we need a dummy data plane cluster.
+    setUpstreamCount(1);
+    setUpstreamProtocol(Http::CodecType::HTTP2);
+    HttpIntegrationTest::initialize();
+    // Register admin port.
+    registerTestServerPorts({});
+    initial_load_success_ = test_server_->counter("runtime.load_success")->value();
+  }
+
+  void acceptXdsConnection() {
+    createXdsConnection();
+    AssertionResult result = xds_connection_->waitForNewStream(*dispatcher_, xds_stream_);
+    RELEASE_ASSERT(result, result.message());
+    xds_stream_->startGrpcStream();
+  }
+
+  std::string getRuntimeKey(const std::string& key) {
+    auto response = IntegrationUtil::makeSingleRequest(
+        lookupPort("admin"), "GET", "/runtime?format=json", "", downstreamProtocol(), version_);
+    EXPECT_TRUE(response->complete());
+    EXPECT_EQ("200", response->headers().getStatusValue());
+    Json::ObjectSharedPtr loader = TestEnvironment::jsonLoadFromString(response->body());
+    auto entries = loader->getObject("entries");
+    if (entries->hasObject(key)) {
+      return entries->getObject(key)->getString("final_value");
+    }
+    return "";
+  }
+
+  void waitforOnConfigUpdatedCount(const int expected_count) {
+    TestUtility::waitForCondition(
```

Comment:
would time_system.waitFor() work here?

Reply:
Awesome didn't know about it! Switched to using that.